app/routes/__init__.py

```diff
@@ -26,27 +21,14 @@ async def version_dependency(
     return version
 
 
-async def update_data(
-    row: db.Model, input_data: Union[BaseModel, Dict[str, Any]]  # type: ignore
-) -> db.Model:  # type: ignore
-    """
-    Merge updated metadata filed with existing fields
-    """
-    if isinstance(input_data, BaseModel):
-        input_data = input_data.dict(skip_defaults=True)
-
-    # Make sure, existing metadata not mentioned in request remain untouched
-    if "metadata" in input_data.keys():
-        metadata = row.metadata
-        metadata.update(input_data["metadata"])
-        input_data["metadata"] = metadata
-
-    # new_row = model.from_orm(row)
-    # new_row.metadata = metadata
+async def version_dependency_form(
+    version: str = Form(..., title="Dataset version", regex=VERSION_REGEX)
+):
 
-    await row.update(**input_data).apply()
+    # if version == "latest":
```

Comment:
Is this something you meant to fill out later?

Reply:
I reviewed this again and I prefer the 307 redirect approach. This is cleaner, since the client will know which latest version was received.